packages/language-server/src/quick-fix.ts

```diff
@@ -12,78 +12,182 @@ import { computeQuickFixStableIdInfo } from "@ui5-language-assistant/xml-views-q
   const documentText = document.getText();
   // We prefer to parse the document again to avoid cache state handling
   const { cst, tokenVector } = parse(documentText);
   const xmlDocAst = buildAst(cst as DocumentCstNode, tokenVector);
   switch (diagnostic.code) {
     case 1000: {
       // non stable id
-      return computeCodeActionForQuickFixStableId({
+      return computeCodeActionsForQuickFixStableId({
         document,
         xmlDocument: xmlDocAst,
         nonStableIdDiagnostic: diagnostic,
+        ui5Model,
       });
     }
     default:
-      return undefined;
+      return [];
   }
 }
 
-function computeCodeActionForQuickFixStableId(opts: {
+function computeCodeActionsForQuickFixStableId(opts: {
   document: TextDocument;
   xmlDocument: XMLDocument;
   nonStableIdDiagnostic: Diagnostic;
-}): CodeAction | undefined {
+  ui5Model: UI5SemanticModel;
+}): CodeAction[] | undefined {
+  let codeActions: CodeAction[] = [];
   const errorOffset = LSPRangeToOffsetRange(
     opts.nonStableIdDiagnostic.range,
     opts.document
   );
 
-  const quickFixStableIdInfo = computeQuickFixStableIdInfo(
-    opts.xmlDocument,
-    errorOffset
-  );
-  if (quickFixStableIdInfo === undefined) {
-    return undefined;
-  }
+  const quickFixStableIdInfo = computeQuickFixStableIdInfo(opts.xmlDocument, [
+    errorOffset,
+  ]);
 
   const replaceRange = offsetRangeToLSPRange(
-    quickFixStableIdInfo.replaceRange,
+    quickFixStableIdInfo[0].replaceRange,
     opts.document
   );
-  return CodeAction.create(
-    QUICK_FIX_STABLE_ID_COMMAND_TITLE,
-    Command.create(
+
+  codeActions.push(
+    CodeAction.create(
       QUICK_FIX_STABLE_ID_COMMAND_TITLE,
-      QUICK_FIX_STABLE_ID_COMMAND,
-      opts.document.uri,
-      replaceRange,
-      quickFixStableIdInfo.newText
-    ),
-    CodeActionKind.QuickFix
+      Command.create(
+        QUICK_FIX_STABLE_ID_COMMAND_TITLE,
+        QUICK_FIX_STABLE_ID_COMMAND,
+        opts.document.uri,
+        opts.document.version,
+        replaceRange,
+        quickFixStableIdInfo[0].newText
+      ),
+      CodeActionKind.QuickFix
+    )
+  );
+
+  const quickFixFileStableIdCodeActions = computeCodeActionsForQuickFixFileStableId(
+    {
+      document: opts.document,
+      xmlDocument: opts.xmlDocument,
+      ui5Model: opts.ui5Model,
+    }
   );
+
+  codeActions = codeActions.concat(quickFixFileStableIdCodeActions);
+
+  return codeActions;
+}
+
+function computeCodeActionsForQuickFixFileStableId(opts: {
+  document: TextDocument;
+  xmlDocument: XMLDocument;
+  ui5Model: UI5SemanticModel;
+}): CodeAction[] {
+  const validators = {
+    document: [],
+    element: [validateNonStableId],
+    attribute: [],
+  };
+
+  const nonStableIdFileIssues = validateXMLView({
+    validators,
+    model: opts.ui5Model,
+    xmlView: opts.xmlDocument,
+  });
+
+  if (nonStableIdFileIssues.length === 1) {
+    return [];
+  }
+
+  const errorsOffset = map(nonStableIdFileIssues, (_) => _.offsetRange);
```

Comment:
instead of checking if some length === 1,
Could we remove the known diagnostic range from `nonStableIdFileIssues` before doing the map? or is it too complicated because we have different kinds of ranges here?